prov/efa/src/efa_device.c

```diff
@@ -178,34 +178,46 @@ int efa_device_list_initialize(void)
 
 	initialized = true;
 
-	ibv_device_list = ibv_get_device_list(&g_device_cnt);
+	ibv_device_list = ibv_get_device_list(&total_device_cnt);
```

Comment:
I probably would have left this as `g_device_cnt` and had a `selected_device_count` or something for the `efa_device_construct()` call, but I guess this works too.

Reply:
Updated. We need both the total EFA device count and the selected device count.